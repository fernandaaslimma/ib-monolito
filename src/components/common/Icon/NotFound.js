import React from "react";

function NotFound() {
  return (
    <svg
      xmlns="http://www.w3.org/2000/svg"
      width="140"
      height="140"
      viewBox="0 0 140 140"
    >
      <g fill="none" fillRule="evenodd">
        <path d="M0 0h140v140H0z" />
        <g stroke="#27445F" strokeWidth="8.27">
          <path strokeLinejoin="round" d="M15.622 21.29v98.323h108.43V21.29z" />
          <path d="M15.622 40.353h108.43" />
        </g>
        <path
          stroke="#F6F9FB"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="18.378"
          d="M91.155 90.091l31.979 29.522"
        />
        <g transform="translate(27.568 24.966)">
          <circle cx="41.351" cy="41.351" r="41.351" fill="#F6F9FB" />
          <path
            stroke="#27445F"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="7.351"
            d="M32.854 33.67l20.258 19.614M52.992 33.186L32.734 52.799"
          />
          <ellipse
            cx="42.062"
            cy="42.042"
            stroke="#27445F"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="8.27"
            rx="31.954"
            ry="31.934"
          />
        </g>
        <g transform="translate(27.568 24.966)">
          <circle cx="41.351" cy="41.351" r="41.351" fill="#F6F9FB" />
          <path
            stroke="#27445F"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="7.351"
            d="M32.854 33.67l20.258 19.614M52.992 33.186L32.734 52.799"
          />
          <ellipse
            cx="42.062"
            cy="42.042"
            stroke="#27445F"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="8.27"
            rx="31.954"
            ry="31.934"
          />
        </g>
        <g
          stroke="#27445F"
          strokeLinecap="round"
          strokeLinejoin="round"
          strokeWidth="8.27"
        >
          <path d="M92.58 90.578l31.013 29.494" />
        </g>
      </g>
    </svg>
  );
}

export default NotFound;
