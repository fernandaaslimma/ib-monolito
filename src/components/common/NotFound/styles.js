import styled, { css } from "styled-components";
import { rem, media } from "../../../styles/tools";

export const Content = styled.div`
  margin-top: ${rem(30)};

  ${media.md(css`
    margin-top: ${rem(30)};
  `)};
`;

export const Wrapper = styled.div`
  span {
    width: 65%;

    ${media.md(css`
      width: 100%;
    `)};
  }
`;
