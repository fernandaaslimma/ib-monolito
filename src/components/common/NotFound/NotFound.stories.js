import React from "react";
import { storiesOf } from "@storybook/react"; // eslint-disable-line
import { withInfo } from "@storybook/addon-info"; // eslint-disable-line

import NotFound from "./NotFound";

storiesOf("common/NotFound", module).add(
  "default",
  withInfo()(() => <NotFound />)
);
