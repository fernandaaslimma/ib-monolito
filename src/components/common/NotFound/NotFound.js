import React from "react";

import { translate } from "../../../utils/i18n";
import { black30, darkBlue, white } from "../../../styles/settings";
import Icon from "../Icon";
import DefaultContent from "../DefaultContent";
import Link from "../Link";
import { isInternetExplorer as isIe } from "../../../utils/getNavigator";
import { Content, Wrapper } from "./styles";

function NotFound() {
  return (
    <Wrapper>
      <DefaultContent
        data-test="NotFound"
        Icon={() => (
          <Icon dataTest="icon-notFound" type="NotFound" color={black30} />
        )}
        primaryText={translate("PAGE_NOT_FOUND")}
        secondaryTexts={[translate("SORRY_PAGE_NOT_FOUND")]}
        isIe={isIe()}
      >
        <Content>
          <Link
            background={darkBlue}
            color={white}
            width={240}
            to="/home"
            dataTest="button-not-found"
            actionSecondary={true}
          >
            {translate("BACK_TO_HOME_SCREEN")}
          </Link>
        </Content>
      </DefaultContent>
    </Wrapper>
  );
}

export default NotFound;
