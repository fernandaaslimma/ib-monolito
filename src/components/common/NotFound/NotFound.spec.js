import React from "react";
import { render, screen, configure } from "@testing-library/react";
import NotFound from "./NotFound";

configure({ testIdAttribute: "data-test" });
jest.mock("../Link", () => {
  const mock = () => <div>test</div>;
  return mock;
});
jest.mock("../../../utils/i18n", () => ({
  translate: id => id,
  getLanguage: () => "en-US",
  isZhCN: () => false
}));

describe("NotFound component", () => {
  it("should match snapshot", () => {
    expect(render(<NotFound />)).toMatchSnapshot();
  });
  it("should render notFound icon", () => {
    render(<NotFound />);
    const icon = screen.getByTestId("icon-notFound");
    expect(icon.getAttribute("color")).toBe("#374654");
  });
});
