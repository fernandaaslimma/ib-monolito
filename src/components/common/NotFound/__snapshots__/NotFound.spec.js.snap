// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-gzVnrw dFUuqt"
      >
        <div
          class="sc-bwzfXH dpzoex"
          data-test="NotFound"
        >
          <div
            class="sc-htpNat iVFZhG"
          >
            <span
              class="sc-bdVaJa iUOSvt"
              color="#374654"
              data-test="icon-notFound"
            >
              <svg
                height="140"
                viewBox="0 0 140 140"
                width="140"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                >
                  <path
                    d="M0 0h140v140H0z"
                  />
                  <g
                    stroke="#27445F"
                    stroke-width="8.27"
                  >
                    <path
                      d="M15.622 21.29v98.323h108.43V21.29z"
                      stroke-linejoin="round"
                    />
                    <path
                      d="M15.622 40.353h108.43"
                    />
                  </g>
                  <path
                    d="M91.155 90.091l31.979 29.522"
                    stroke="#F6F9FB"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="18.378"
                  />
                  <g
                    transform="translate(27.568 24.966)"
                  >
                    <circle
                      cx="41.351"
                      cy="41.351"
                      fill="#F6F9FB"
                      r="41.351"
                    />
                    <path
                      d="M32.854 33.67l20.258 19.614M52.992 33.186L32.734 52.799"
                      stroke="#27445F"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="7.351"
                    />
                    <ellipse
                      cx="42.062"
                      cy="42.042"
                      rx="31.954"
                      ry="31.934"
                      stroke="#27445F"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="8.27"
                    />
                  </g>
                  <g
                    transform="translate(27.568 24.966)"
                  >
                    <circle
                      cx="41.351"
                      cy="41.351"
                      fill="#F6F9FB"
                      r="41.351"
                    />
                    <path
                      d="M32.854 33.67l20.258 19.614M52.992 33.186L32.734 52.799"
                      stroke="#27445F"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="7.351"
                    />
                    <ellipse
                      cx="42.062"
                      cy="42.042"
                      rx="31.954"
                      ry="31.934"
                      stroke="#27445F"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="8.27"
                    />
                  </g>
                  <g
                    stroke="#27445F"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="8.27"
                  >
                    <path
                      d="M92.58 90.578l31.013 29.494"
                    />
                  </g>
                </g>
              </svg>
            </span>
          </div>
          <h3
            class="sc-bxivhb CsUg"
            data-test="NotFoundTitle"
          >
            PAGE_NOT_FOUND
          </h3>
          <p
            class="sc-EHOje jifphG"
            data-test="NotFoundSubTitle"
          >
            SORRY_PAGE_NOT_FOUND
          </p>
          <div
            class="sc-bZQynM brfISi"
          >
            <div>
              test
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-gzVnrw dFUuqt"
    >
      <div
        class="sc-bwzfXH dpzoex"
        data-test="NotFound"
      >
        <div
          class="sc-htpNat iVFZhG"
        >
          <span
            class="sc-bdVaJa iUOSvt"
            color="#374654"
            data-test="icon-notFound"
          >
            <svg
              height="140"
              viewBox="0 0 140 140"
              width="140"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="none"
                fill-rule="evenodd"
              >
                <path
                  d="M0 0h140v140H0z"
                />
                <g
                  stroke="#27445F"
                  stroke-width="8.27"
                >
                  <path
                    d="M15.622 21.29v98.323h108.43V21.29z"
                    stroke-linejoin="round"
                  />
                  <path
                    d="M15.622 40.353h108.43"
                  />
                </g>
                <path
                  d="M91.155 90.091l31.979 29.522"
                  stroke="#F6F9FB"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="18.378"
                />
                <g
                  transform="translate(27.568 24.966)"
                >
                  <circle
                    cx="41.351"
                    cy="41.351"
                    fill="#F6F9FB"
                    r="41.351"
                  />
                  <path
                    d="M32.854 33.67l20.258 19.614M52.992 33.186L32.734 52.799"
                    stroke="#27445F"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="7.351"
                  />
                  <ellipse
                    cx="42.062"
                    cy="42.042"
                    rx="31.954"
                    ry="31.934"
                    stroke="#27445F"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="8.27"
                  />
                </g>
                <g
                  transform="translate(27.568 24.966)"
                >
                  <circle
                    cx="41.351"
                    cy="41.351"
                    fill="#F6F9FB"
                    r="41.351"
                  />
                  <path
                    d="M32.854 33.67l20.258 19.614M52.992 33.186L32.734 52.799"
                    stroke="#27445F"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="7.351"
                  />
                  <ellipse
                    cx="42.062"
                    cy="42.042"
                    rx="31.954"
                    ry="31.934"
                    stroke="#27445F"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="8.27"
                  />
                </g>
                <g
                  stroke="#27445F"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="8.27"
                >
                  <path
                    d="M92.58 90.578l31.013 29.494"
                  />
                </g>
              </g>
            </svg>
          </span>
        </div>
        <h3
          class="sc-bxivhb CsUg"
          data-test="NotFoundTitle"
        >
          PAGE_NOT_FOUND
        </h3>
        <p
          class="sc-EHOje jifphG"
          data-test="NotFoundSubTitle"
        >
          SORRY_PAGE_NOT_FOUND
        </p>
        <div
          class="sc-bZQynM brfISi"
        >
          <div>
            test
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
